controllers/servicediscovery/servicediscovery_controller.go

```diff
@@ -173,12 +172,14 @@ func (r *ServiceDiscoveryReconciler) Reconcile(request reconcile.Request) (recon
 	if err != nil {
 		return reconcile.Result{}, err
 	}
-	err = updateDNSCustomConfigMap(r.client, r.k8sClientSet, instance, reqLogger)
-	if errors.IsNotFound(err) {
+	if instance.Spec.CoreDNSCustomConfigName != "" {
```

Comment:
This changes the default behaviour, doesn’t it? Currently we look for `coredns-custom`, update that if it’s present, or call `updateDNSConfigMap()` if it isn’t; with this patch we’ll only do that if the custom config name is set. Does that matter?

Reply:
Yes, I wanted to check with Jakob before making the change. If custom config is set, we will create if not already present.